io/eolearn/io/processing_api.py

```diff
@@ -218,40 +226,36 @@ def __init__(self, data_source, size=None, resolution=None, bands_feature=None,
         :param bands_dtype: dtype of the bands array
         :type bands_dtype: type
         :param single_scene: If true, the service will compute a single image for the given time interval using
-                             mosaicking.
+            mosaicking.
         :type single_scene: bool
         :param mosaicking_order: Mosaicking order, which has to be either 'mostRecent', 'leastRecent' or 'leastCC'.
         :type mosaicking_order: str
         :param aux_request_args: a dictionary with auxiliary information for the input_data part of the SH request
         :type aux_request_args: dict
+        :param data_source: A deprecated alternative to data_collection
+        :type data_source: DataCollection
         """
         super().__init__(
-            data_source=data_source, size=size, resolution=resolution, cache_folder=cache_folder, config=config,
-            max_threads=max_threads
+            data_collection=data_collection, size=size, resolution=resolution, cache_folder=cache_folder, config=config,
+            max_threads=max_threads, data_source=data_source
         )
-
-        self.data_source = data_source
+        self.evalscript = evalscript
         self.maxcc = maxcc
         self.time_difference = dt.timedelta(seconds=1) if time_difference is None else time_difference
         self.single_scene = single_scene
         self.bands_dtype = bands_dtype
         self.mosaicking_order = mosaicking_order
         self.aux_request_args = aux_request_args
 
+        self.bands_feature = next(self._parse_features(bands_feature, allowed_feature_types=[FeatureType.DATA])()) \
+            if bands_feature else None
         self.requested_bands = {}
 
         if bands_feature:
-            bands_feature = next(self._parse_features(bands_feature, allowed_feature_types=[FeatureType.DATA])())
-
-            if not bands and data_source in [DataSource.SENTINEL2_L1C, DataSource.SENTINEL2_L2A]:
-                bands = data_source.bands()
-            elif data_source.is_custom() and not bands:
-                raise ValueError("For custom data sources 'bands' must be provided as an argument.")
-
+            if not bands:
+                bands = data_collection.bands
```

Comment:
I don't think that this ensures `bands` to be empty list. Some collections have no default bands specified.

Reply:
If bands are not defined for a given collection this will raise an error. Note that `DataCollection.bands` is actually a property -> [implementation](https://github.com/sentinel-hub/sentinelhub-py/blob/develop/sentinelhub/data_collections.py#L475). It is a bit sneaky but also very convenient to use.